dbt_subprojects/hourly_spellbook/models/_sector/perpetual/projects/avantis/avantis_v1_base_perpetual_trades.sql

```diff
@@ -0,0 +1,201 @@
+{{ config(
+    schema = 'avantis_base',
+    alias = 'perpetual_trades',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["base"]\',
+                                "project",
+                                "avantis",
+                                \'["princi"]\') }}'
+)
+}}
+
+-- {% set project_start_date = '2024-01-01' %}
+
+WITH trading_events AS (
+    -- OpenLimitPlaced events
+    SELECT
+        evt_block_time,
+        evt_block_number,
+        evt_tx_hash,
+        evt_tx_from,
+        evt_tx_to,
+        evt_index,
+        contract_address,
+        trader,
+        pairIndex,
+        index as position_size,
+        openPrice / 1e8 as price_usd,
+        executionFee / 1e18 as fee_usd,
+        isBuy,
+        'limit_order' as event_type
+    FROM delta_prod.avantis_base.Trading_evt_OpenLimitPlaced
+
```

Comment:
Please remember to add incremental filters where needed.
```suggestion
    {% if is_incremental() %}
    WHERE {{ incremental_predicate('evt_block_time') }}
    {% endif %}

```